components/scream/src/dynamics/homme/physics_dynamics_remapper.hpp

```diff
@@ -272,133 +314,496 @@ do_registration_ends ()
   if (this->m_num_bound_fields==this->m_num_registered_fields) {
     setup_boundary_exchange ();
   }
+
+  // Precompute fields needed on device during remapper
+  initialize_device_variables();
 }
 
 template<typename RealType>
 void PhysicsDynamicsRemapper<RealType>::
-do_remap_fwd() const
+initialize_device_variables()
 {
-  using pack_type = ekat::Pack<RealType,SCREAM_PACK_SIZE>;
-  using small_pack_type = ekat::Pack<RealType,SCREAM_SMALL_PACK_SIZE>;
+  const int num_fields = m_phys.size();
+
+  phys_ptrs      = decltype(phys_ptrs)      ("phys_ptrs",      num_fields);
+  phys_dims      = decltype(phys_dims)      ("phys_dims",      num_fields, 4);
+  phys_dims_size = decltype(phys_dims_size) ("phys_dims_size", num_fields);
+  phys_layout    = decltype(phys_layout)    ("phys_layout",    num_fields);
+
+  dyn_ptrs   = decltype(dyn_ptrs)   ("dyn_ptrs",   num_fields);
+  dyn_dims   = decltype(dyn_dims)   ("dyn_dims",   num_fields, 6);
+  dyn_layout = decltype(dyn_layout) ("dyn_layout", num_fields);
+
+  pack_alloc_property = decltype(pack_alloc_property) ("phys_pack_alloc_property", num_fields);
+  itl_fwd             = decltype(itl_fwd)             ("itl_fwd",                  num_fields);
+  itl_bwd             = decltype(itl_bwd)             ("itl_bwd",                  num_fields);
+  ordering            = decltype(ordering)            ("ordering_new",             num_fields, 2);
```

Comment:
Somewhat similar to phys_dims: would it be beneficial to have ordering be a `view_1d<Kokkos::pair<int,int>>`? I suspect the answer is "yes". Perhaps `view<Int*[2]>` would have similar benefits, though I still suspect the pair approach would yield better register usage.

Reply:
This no longer exists (removed tensor3d).